patches/api/0225-Add-a-way-to-get-translation-keys-for-blocks-entitie.patch

```diff
@@ -78,36 +152,96 @@ index 786b8011e98b2fe93cc2418d624f6350ede62d90..024deba760c41787190d20e4ac5c5419
+index 8eb0497c81744874809ebc4bc2e28b128e66a926..b277034fee2d4f38c40713842d14a8f6dde757aa 100644
+--- a/src/main/java/org/bukkit/enchantments/Enchantment.java
++++ b/src/main/java/org/bukkit/enchantments/Enchantment.java
+@@ -12,7 +12,7 @@ import org.jetbrains.annotations.Nullable;
+ /**
+  * The various type of enchantments that may be added to armour or weapons
+  */
+-public abstract class Enchantment implements Keyed {
++public abstract class Enchantment implements Keyed, net.kyori.adventure.translation.Translatable { // Paper - Adventure translations
+     /**
+      * Provides protection against environmental damage
+      */
+diff --git a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+index 4d5f0837bd0e02a30c943d8969fb6b13452322e0..a39f9c078f42451bd122f3e3729d10ca299bee5f 100644
+--- a/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
++++ b/src/main/java/org/bukkit/enchantments/EnchantmentWrapper.java
+@@ -69,5 +69,10 @@ public class EnchantmentWrapper extends Enchantment {
+     public net.kyori.adventure.text.Component displayName(int level) {
+         return getEnchantment().displayName(level);
+     }
++
++    @Override
++    public @NotNull String translationKey() {
++        return getEnchantment().translationKey();
++    }
+     // Paper end
+ }
 diff --git a/src/main/java/org/bukkit/entity/EntityType.java b/src/main/java/org/bukkit/entity/EntityType.java
-index 9be5371c7f398d0ec8241403661415ff40661067..f415b61b0d4b57e1557aaf240a0f2ad5915035fc 100644
+index 9be5371c7f398d0ec8241403661415ff40661067..d36d314383713bac3b11f18d95b0809dce3cd6e0 100644
 --- a/src/main/java/org/bukkit/entity/EntityType.java
 +++ b/src/main/java/org/bukkit/entity/EntityType.java
-@@ -419,4 +419,15 @@ public enum EntityType implements Keyed {
+@@ -20,7 +20,7 @@ import org.jetbrains.annotations.Contract;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
+-public enum EntityType implements Keyed {
++public enum EntityType implements Keyed, net.kyori.adventure.translation.Translatable { // Paper - translatable
+ 
+     // These strings MUST match the strings in nms.EntityTypes and are case sensitive.
+     /**
+@@ -419,4 +419,27 @@ public enum EntityType implements Keyed {
      public boolean isAlive() {
          return living;
      }
-+
++    // Paper start
 +    /**
 +     * Return the translation key for the EntityType, so the client can translate it into the active
 +     * locale when using a TranslatableComponent.<br>
 +     * This is <code>null</code>, when the EntityType isn't known to NMS (custom entities)
 +     * @return the translation key
++     * @deprecated use {@link #translationKey()}
 +     */
++    @Deprecated
 +    @Nullable
-+    String getTranslationKey() {
++    public String getTranslationKey() {
++        return org.bukkit.Bukkit.getUnsafe().getTranslationKey(this);
++    }
++
++    /**
```

Comment:
should have a `{@inheritDoc}` i think

Reply:
nah, not nessecary, it auto inherits parts of the doc it doesnt have, like main description, return value, etc.